tests/extension/oneapi_auto_local_range/auto_local_range.cpp

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "../../common/common.h"
+#include <type_traits>
+
+namespace auto_local_range::tests {
+
+#ifdef SYCL_EXT_ONEAPI_AUTO_LOCAL_RANGE
+
+template <int Dimensions>
+static void check_auto_range_result_type() {
+  static_assert(1 <= Dimensions && Dimensions <= 3);
+  using F = decltype(sycl::ext::oneapi::experimental::auto_range<Dimensions>);
+  static_assert(std::is_same_v<typename std::invoke_result<F>::type,
+                               sycl::range<Dimensions>>);
+}
+
+template <size_t... Dims>
+static void check_auto_range() {
+  constexpr int Dimensions = sizeof...(Dims);
+  static_assert(1 <= Dimensions && Dimensions <= 3);
+
+  sycl::queue q;
+  sycl::range<Dimensions> N{Dims...};
+  sycl::buffer<int, Dimensions> input_buffer{N};
+  sycl::buffer<int, Dimensions> output_buffer{N};
+
+  {
+    auto input = input_buffer.get_host_access();
+    int i = 0;
+    for (auto& it : input) {
+      it = i + 1;
+      i++;
```

Comment:
```suggestion
      it = ++i;
```
Can we simplify the code like this?
The whole loop looks like an implementation of `std::iota`.